Lib/test/test_buffer.py

```diff
@@ -4591,6 +4591,23 @@ def test_c_buffer_invalid_flags(self):
         self.assertRaises(SystemError, buf.__buffer__, PyBUF_READ)
         self.assertRaises(SystemError, buf.__buffer__, PyBUF_WRITE)
 
+    @unittest.skipIf(_testcapi is None, "requires _testcapi")
+    def test_c_fill_buffer_invalid_flags(self):
+        # PyBuffer_FillInfo
+        self.assertRaises(SystemError, _testcapi.buffer_fill_info,
+                          "abc", 0, PyBUF_READ)
+        self.assertRaises(SystemError, _testcapi.buffer_fill_info,
+                          "abc", 0, PyBUF_WRITE)
+
+    @unittest.skipIf(_testcapi is None, "requires _testcapi")
+    def test_c_fill_buffer_readonly_and_writable(self):
+        self.assertEqual(
+            str(_testcapi.buffer_fill_info("abc", 1, PyBUF_SIMPLE), "utf8"),
```

Comment:
Why not use `bytes()`.

You can also test it with PyBUF_WRITABLE flag and readonly=0. But you need to fill the buffer from a bytearray for safety. Test also that the resulting memoryview is actually readonly or writeable.